src/client/components/Products.tsx

```diff
@@ -1,4 +1,10 @@
+import { useState } from "react";
+import Modal from "./Modal";
+
 export default function Products() {
+
+  const [showModal, setShowModal] = useState(false);
+
```

Comment:
Great Job Nikita , I was thinking it might be helpful to add another useState variable here ( Paid ), this will be helpful in deciding what to show after the user completes the payment. ex: if instead of paying maybe a certain document will be downloaded. This will need to be passed to the Modal component as well.

Reply:
Yeah, Thank you for the review! I will add paid state too!